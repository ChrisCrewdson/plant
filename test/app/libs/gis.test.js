const gis = require('../../../app/libs/gis');
const seamless = require('seamless-immutable').static;

describe('/app/libs/gis', () => {
  describe('scaling to canvas', () => {
    test('should scale zero plants', () => {
      const width = 700;
      const immutablePlants = seamless.from({});
      const scaledPlants = gis.scaleToCanvas(immutablePlants, width);
      expect(seamless.isImmutable(scaledPlants.plants)).toBe(true);
      expect(scaledPlants).toMatchSnapshot();
    });

    test('should scale a single plant', () => {
      const plants = {
        1: {
          _id: '1',
          title: 'Title 1',
          loc: {
            coordinates: [10, 20],
          },
        },
      };
      const width = 700;
      const immutablePlants = seamless.from(plants);
      const scaledPlants = gis.scaleToCanvas(immutablePlants, width);
      expect(seamless.isImmutable(scaledPlants.plants)).toBe(true);
      expect(scaledPlants).toMatchSnapshot();
    });

    test('should scale two plants on the same latitude', () => {
      const plants = {
        1: {
          _id: '1',
          title: 'Title 1',
          loc: {
            coordinates: [10, 20],
          },
        },
        2: {
          _id: '2',
          title: 'Title 2',
          loc: {
            coordinates: [11, 20],
          },
        },
      };
      const width = 700;
      const immutablePlants = seamless.from(plants);
      const scaledPlants = gis.scaleToCanvas(immutablePlants, width);
      expect(seamless.isImmutable(scaledPlants.plants)).toBe(true);
      expect(scaledPlants).toMatchSnapshot();
    });

    test('should scale two plants on the same longitude', () => {
      const plants = {
        1: {
          _id: '1',
          title: 'Title 1',
          loc: {
            coordinates: [10, 20],
          },
        },
        2: {
          _id: '2',
          title: 'Title 2',
          loc: {
            coordinates: [10, 21],
          },
        },
      };
      const width = 700;
      const immutablePlants = seamless.from(plants);
      const scaledPlants = gis.scaleToCanvas(immutablePlants, width);
      expect(seamless.isImmutable(scaledPlants.plants)).toBe(true);
      expect(scaledPlants).toMatchSnapshot();
    });
  });
});
