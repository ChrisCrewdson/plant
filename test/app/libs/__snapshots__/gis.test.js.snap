// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/app/libs/gis scaling to canvas should scale a single plant 1`] = `
Object {
  "canvasHeight": 700,
  "plants": Object {
    "1": Object {
      "_id": "1",
      "title": "Title 1",
      "x": 350,
      "y": 350,
    },
  },
}
`;

exports[`/app/libs/gis scaling to canvas should scale two plants on the same latitude 1`] = `
Object {
  "canvasHeight": 700,
  "plants": Object {
    "1": Object {
      "_id": "1",
      "title": "Title 1",
      "x": 50,
      "y": 50,
    },
    "2": Object {
      "_id": "2",
      "title": "Title 2",
      "x": 650,
      "y": 50,
    },
  },
}
`;

exports[`/app/libs/gis scaling to canvas should scale two plants on the same longitude 1`] = `
Object {
  "canvasHeight": 700,
  "plants": Object {
    "1": Object {
      "_id": "1",
      "title": "Title 1",
      "x": 50,
      "y": 50,
    },
    "2": Object {
      "_id": "2",
      "title": "Title 2",
      "x": 50,
      "y": 650,
    },
  },
}
`;

exports[`/app/libs/gis scaling to canvas should scale zero plants 1`] = `
Object {
  "canvasHeight": 0,
  "plants": Object {},
}
`;
